mailpoet/tests/integration/AdminPages/HelpTest.php

```diff
@@ -0,0 +1,107 @@
+<?php declare(strict_types=1);
+
+namespace integration\AdminPages;
+
+use MailPoet\AdminPages\PageRenderer;
+use MailPoet\AdminPages\Pages\Help;
+use MailPoet\Cron\CronHelper;
+use MailPoet\Entities\NewsletterEntity;
+use MailPoet\Entities\ScheduledTaskEntity;
+use MailPoet\Entities\SendingQueueEntity;
+use MailPoet\Helpscout\Beacon;
+use MailPoet\Newsletter\Sending\ScheduledTasksRepository;
+use MailPoet\Newsletter\Sending\SendingQueuesRepository;
+use MailPoet\Newsletter\Url;
+use MailPoet\Services\Bridge;
+use MailPoet\Tasks\Sending;
+use MailPoet\Test\DataFactories\Newsletter;
+use MailPoet\Test\DataFactories\ScheduledTask as ScheduledTaskFactory;
+use MailPoetVendor\Carbon\Carbon;
+
+class HelpTest extends \MailPoetTest {
+
+  /** @var ScheduledTaskFactory */
+  private $scheduledTaskFactory;
+
+  /*** @var SendingQueuesRepository */
+  private $sendingQueuesRepository;
+
+  /**
+   * @var Help
+   */
+  private $helpPage;
+
+  public function _before() {
+    parent::_before();
+    $this->cleanup();
+    $this->scheduledTaskFactory = new ScheduledTaskFactory();
+    $this->sendingQueuesRepository = $this->diContainer->get(SendingQueuesRepository::class);
+
+    $this->helpPage = new Help(
+      $this->diContainer->get(PageRenderer::class),
+      $this->diContainer->get(CronHelper::class),
+      $this->diContainer->get(Beacon::class),
+      $this->diContainer->get(Bridge::class),
+      $this->diContainer->get(ScheduledTasksRepository::class),
+      $this->sendingQueuesRepository,
+      $this->diContainer->get(Url::class)
+    );
+  }
+
+  public function testItFetchesNewsletterDataForSendingTasks() {
+    $task = $this->scheduledTaskFactory->create(
+      Sending::TASK_TYPE,
+      ScheduledTaskEntity::STATUS_SCHEDULED,
+      Carbon::now()->addDay()
+    );
+    $newsletter = (new Newsletter())
+      ->withSubject('Rendered Subject')
+      ->create();
+    $this->createNewSendingQueue($task, $newsletter);
+    $data = $this->helpPage->buildTaskData($task);
+    expect($data['newsletter']['newsletter_id'])->equals(1);
+    expect($data['newsletter']['queue_id'])->equals(1);
+    expect($data['newsletter']['subject'])->equals('Rendered Subject');
+    expect($data['newsletter']['preview_url'])->notEmpty();
+  }
+
+  public function testItDoesNotFailForSendingTaskWithMissingNewsletterInconsistentData() {
+    $task = $this->scheduledTaskFactory->create(
+      Sending::TASK_TYPE,
+      ScheduledTaskEntity::STATUS_SCHEDULED,
+      Carbon::now()->addDay()
+    );
+    $data = $this->helpPage->buildTaskData($task);
+    expect($data['newsletter']['newsletter_id'])->equals(null);
+    expect($data['newsletter']['queue_id'])->equals(null);
+    expect($data['newsletter']['subject'])->equals(null);
+    expect($data['newsletter']['preview_url'])->equals(null);
+  }
+
+  public function createNewSendingQueue(?ScheduledTaskEntity $task, ?NewsletterEntity $newsletter, $renderedSubject = null) {
```

Comment:
I agree that this can be private and basically data factories defined in [another PR](https://github.com/mailpoet/mailpoet/pull/4313/files#diff-19e3f169ce739f526e5702b016ea72e09f52fde590cf7fc91a4148753b6128ecR11) can be used here as well, but this method is assigning `$task` to `$queue` as well, see 
```  
   if ($task instanceof ScheduledTaskEntity) {
      $queue->setTask($task);
    }
```

Reply:
I'm not sure I follow why assigning `$task` to `$queue` as well is a problem in this particular case and it prevents using `withSendingQueue()` or `withScheduledQueue()`. We can always get the `$task` from the `$newsletter` and we might modify the data factory if we need to set specific options for the sending queue or task that are not possible with the current interface. But I don't think this is important enough and I'm code with leaving the code as is.